File: android/src/main/java/expo/modules/gaodemap/overlays/MultiPointViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * MultiPoint 视图 Module
 */
class MultiPointViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("MultiPointView")

    View(MultiPointView::class) {
      Events("onPress")
      
      Prop<List<Map<String, Any>>>("points") { view: MultiPointView, points ->
        view.setPoints(points)
      }
    }
  }
}

File: ios/overlays/MultiPointViewModule.swift
import ExpoModulesCore

public class MultiPointViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("MultiPointView")
        
        View(MultiPointView.self) {
            Events("onPress")
            
            Prop("points") { (view: MultiPointView, points: [[String: Any]]) in
                view.setPoints(points)
            }
        }
    }
}